### PR TITLE
Keypoint loss refactor

### PR DESCRIPTION
**Pull request by kozlov721**
Refactor of the keypoint loss.
Lays the groundwork for further refactoring to split into separate losses for keypoints and boxes.

### CODE REVIEW

File: src/luxonis_train/utils/losses/keypoint_box_loss.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    ):
+        super().__init__()
+
+        self.n_classes = n_classes
+
+        self.n_keypoints: int = cast(int, head_attributes.get("n_keypoints"))
+        self.n_anchors: int = cast(int, head_attributes.get("n_anchors"))
+        self.num_heads: int = cast(int, head_attributes.get("num_heads"))
+        self.anchors: Tensor = cast(Tensor, head_attributes.get("anchors"))
+        if self.num_heads == 3:
```
Comment (tersekmatija) [inline src/luxonis_train/utils/losses/keypoint_box_loss.py:38]: Should this be configurable / argument?

File: src/luxonis_train/utils/losses/keypoint_box_loss.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+        self.iou_ratio = iou_ratio
+        self.box_weight = box_weight
+        self.kpt_weight = kpt_weight
+        self.cls_weight = cls_weight
+        self.obj_weight = obj_weight
+        self.kptv_weight = kptv_weight
+        self.anchor_t = anchor_t
+
+        self.bias = 0.5
```
Comment (tersekmatija) [inline src/luxonis_train/utils/losses/keypoint_box_loss.py:51]: Same, should be an argument perhaps?

File: src/luxonis_train/utils/losses/keypoint_box_loss.py [hunk 3]
```diff
@@ -0,0 +1,352 @@
+from collections.abc import Mapping
+from typing import Any, Dict, Final, List, Tuple, cast
+
+import torch
+from torch import Tensor, nn
+
+from luxonis_train.utils.boxutils import bbox_iou
+from luxonis_train.utils.losses.common import BCEWithLogitsLoss
+
+BOX_OFFSET: Final[int] = 5
```
Comment (tersekmatija) [inline src/luxonis_train/utils/losses/keypoint_box_loss.py:10]: If we add support for rotated bounding boxes, would offset change or would we treat angle as "additional variable"?

File: src/luxonis_train/utils/losses/keypoint_box_loss.py (hunk 1 above)
Comment (kozlov721) [reply src/luxonis_train/utils/losses/keypoint_box_loss.py:38]: This is configurable, but in the head.

Comment (tersekmatija) [reply src/luxonis_train/utils/losses/keypoint_box_loss.py:38]: Sorry, I meant the whole "if statement", or more specifically `self.balance`.
